File: AndroidStudioProjects/ListView20201126/app/src/main/java/ac/baekseok/listview20201126/MainActivity.java
package ac.baekseok.listview20201126;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;


public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            setContentView(R.layout.activity_main);
            setTitle("리스트뷰 테스트 엄진하");
            //리스트뷰에 나열될 항목의 글자를 배열로 선언 final onCreate()
            final String[] mid={"ICT개론","IOT실습","해킹이론","안드로이드프로그래밍",
                    "파이썬","인공지능개론","VR실습","빅데이터","모바일 웹"};
            //추가했던 리스트뷰
            ListView list=(ListView)findViewById(R.id.listView1);
            //리스트뷰에 형식 단순리스트타입, 나열되는 글자를 지정하기 위한 ArrayAdapter를 추가
            ArrayAdapter<String> adapter=new ArrayAdapter<String>(this,
                android.R.layout.simple_list_item_1,mid);
        //매개변수 리스트의 단순리스트 형식,나열글자
        list.setAdapter(adapter);//형식,나열글자가 적용됌 리스트뷰에 설정된 ArrayAdapter를 적용
        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                //position이라는 매개변수는 list의 아이템중에 몇번째를 저장하고 있는지를 의미
                if(position==0){
                    //화면 Second.java로 이동, 인텐트 이동
                    Intent intent= new Intent(getApplicationContext(),Second.class);
                    startActivity(intent);
                }
                else if(position==1){
                    Intent intent= new Intent(getApplicationContext(),Third.class);
                    startActivity(intent);
                }
            }
        });
    }//onCreate
}//MainActivity

File: mango_contents/app/src/main/java/bu/ac/kr/mango_contents/MainActivity.kt
package bu.ac.kr.mango_contents

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private val items = mutableListOf<ContentsModel>()


    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/XRoMziImmYCC",
                "https://mp-seoul-image-production-s3.mangoplate.com/331247/60039_1596540913676_34054?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "뉴욕택시"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/mfwd78vEtS",
                "https://mp-seoul-image-production-s3.mangoplate.com/13936/704481_1492338122499_12072?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "파씨오네"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/fPi6OZ3r-u",
                "https://mp-seoul-image-production-s3.mangoplate.com/516929_1525669069306497.jpg?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
            "비야게레로"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/08OQhkbuBywp",
            "https://mp-seoul-image-production-s3.mangoplate.com/806970_1500088411444579.jpg?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
            "팬케이크"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/XRoMziImmYCC",
                "https://mp-seoul-image-production-s3.mangoplate.com/331247/60039_1596540913676_34054?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "뉴욕택시"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/mfwd78vEtS",
                "https://mp-seoul-image-production-s3.mangoplate.com/13936/704481_1492338122499_12072?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "파씨오네"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/fPi6OZ3r-u",
                "https://mp-seoul-image-production-s3.mangoplate.com/516929_1525669069306497.jpg?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "비야게레로"
            )
        )
        items.add(
            ContentsModel(
                "https://www.mangoplate.com/restaurants/08OQhkbuBywp",
                "https://mp-seoul-image-production-s3.mangoplate.com/806970_1500088411444579.jpg?fit=around|512:512&crop=512:512;*,*&output-format=jpg&output-quality=80",
                "팬케이크"
            )
        )

        val recyclerview = findViewById<RecyclerView>(R.id.rv)
        val rvAdapter = RVAdapter(baseContext,items)
        recyclerview.adapter = rvAdapter

        rvAdapter.itemClick= object : RVAdapter.ItemClick{
            override fun onClick(view : View, position:Int){

                val intent = Intent(baseContext,ViewActivity::class.java)
                intent.putExtra("url",items[position].url)
                intent.putExtra("title",items[position].titleText)
                intent.putExtra("imageUrl",items[position].ImageUrl)
                startActivity(intent)


            }
        }
        recyclerview.layoutManager = GridLayoutManager(this,2)

    }
}